scripts/data/generate_data.py

```diff
@@ -20,23 +23,34 @@
 
 FAST = False
 
+RETRIES = 3
+DEPLAY = 2
+
 
 def request_rpc(method: str, params: list):
     """Makes a JSON-RPC call to a Bitcoin API endpoint.
-    Uses environment variables BITCOIN_RPC and USERPWD
-    or the default public endpoint if those variables are not set.
+    Retries the request a specified number of times before failing.
 
+    :param retries: Number of retry attempts before raising an exception.
+    :param delay: Delay between retries in seconds.
     :return: parsed JSON result as Python object
     """
     url = BITCOIN_RPC or DEFAULT_URL
     auth = tuple(USERPWD.split(":")) if USERPWD else None
     headers = {"content-type": "application/json"}
     payload = {"jsonrpc": "2.0", "method": method, "params": params, "id": 0}
-    res = requests.post(url, auth=auth, headers=headers, json=payload)
-    try:
-        return res.json()["result"]
-    except Exception:
-        raise ConnectionError(f"Unexpected RPC response:\n{res.text}")
+
+    for attempt in range(RETRIES):
+        try:
+            res = requests.post(url, auth=auth, headers=headers, json=payload)
+            return res.json()["result"]
+        except Exception:
+            if attempt < RETRIES - 1:
+                time.sleep(DEPLAY)  # Wait before retrying
```

Comment:
Print a message "Connection error: {res.text}, will retry in {DELAY}s" before sleep.